tests/test_taskmanager.py

```diff
@@ -0,0 +1,968 @@
+
+from django.contrib.auth.models import User
+
+from mock import Mock, PropertyMock, patch
+
+from kobo.client.constants import TASK_STATES
+from kobo.exceptions import ShutdownException
+from kobo.hub.models import Arch, Channel, Task, Worker
+from kobo.worker import TaskBase
+from kobo.worker.task import FailTaskException
+from kobo.worker.taskmanager import TaskManager, TaskContainer
+from six.moves.xmlrpc_client import ProtocolError
+
+from .rpc import HubProxyMock, RpcServiceMock
+from .utils import DjangoRunner
+
+runner = DjangoRunner()
+setup_module = runner.start
+teardown_module = runner.stop
+
+
+class DummyTask(TaskBase):
+
+    def run(self):
+        self.result = True
+
+    @classmethod
+    def cleanup(cls, hub, conf, task_info):
+        pass
+
+    @classmethod
+    def notification(cls, hub, conf, task_info):
+        pass
+
+
+class DummyForegroundTask(DummyTask):
+    enabled = True
+    arches = ['testarch']
+    channels = ['testchannel']
+    exclusive = False
+    foreground = True
+    priority = 10
+    weight = 1.0
+
+
+class DummyForkTask(DummyTask):
+    enabled = True
+    arches = ['testarch']
+    channels = ['testchannel']
+    exclusive = False
+    foreground = False
+    priority = 10
+    weight = 1.0
+
+
+class DummyHeavyTask(DummyTask):
+    enabled = True
+    arches = ['testarch']
+    channels = ['testchannel']
+    exclusive = False
+    foreground = False
+    priority = 10
+    weight = 100.0
+
+
+class DummyFailTask(DummyTask):
+    enabled = True
+    arches = ['testarch']
+    channels = ['testchannel']
+    exclusive = False
+    foreground = True
+    priority = 10
+    weight = 1.0
+
+    def run(self):
+        raise FailTaskException()
+
+
+class DummyExceptionTask(DummyTask):
+    enabled = True
+    arches = ['testarch']
+    channels = ['testchannel']
+    exclusive = False
+    foreground = True
+    priority = 10
+    weight = 1.0
+
+    def run(self):
+        raise Exception()
+
+
+class TestTaskManager(django.test.TestCase):
+
+    def setUp(self):
+        super(TestTaskManager, self).setUp()
+
+        TaskContainer.register_plugin(DummyForegroundTask)
+        TaskContainer.register_plugin(DummyForkTask)
+        TaskContainer.register_plugin(DummyHeavyTask)
+        TaskContainer.register_plugin(DummyFailTask)
+        TaskContainer.register_plugin(DummyExceptionTask)
+
+        user = User.objects.create(username='testuser')
+        arch = Arch.objects.create(name='testarch')
+        channel = Channel.objects.create(name='testchannel')
+
+        w = Worker.objects.create(
+            worker_key='mock-worker',
+            name='mock-worker',
+        )
+
+        w.arches.add(arch)
+        w.channels.add(channel)
+        w.save()
+
+        self._arch = arch
+        self._channel = channel
+        self._user = user
+        self._worker = RpcServiceMock(w)
+
+    @unittest.skip('The code has a bug, it must be fixed to get this test working.')
```

Comment:
Did you try using an xfail instead of a skip?

skip has the problem that someone can fix the bug and forget to update the test, or not notice that there's an existing test which would cover the bug... which ultimately could result in the bug being re-introduced.  With xfail you would force that whoever fixes the bug must also update this test at the same time to remove the xfail.

Reply:
Also, can you please elaborate a bit what the bug is?  Ideally referencing an issue number.

As this just says "The code has a bug", and the CI will not even *try* to run the test under any circumstances, I can't see how anyone could know when it's time to remove this skip.

Reply:
Issue #68 created.